modules/cli/src/main/scala/scala/cli/commands/Publish.scala

```diff
@@ -270,43 +315,129 @@ object Publish extends ScalaCommand[PublishOptions] {
       }
       .toSeq
 
+    val docJarEntries = docJarOpt
+      .map { docJar =>
+        (basePath / s"$fullName-$ver-javadoc.jar") -> Content.File(docJar.toNIO)
+      }
+      .toSeq
+
     // TODO version listings, …
-    FileSet(mainEntries ++ sourceJarEntries)
+    FileSet(mainEntries ++ sourceJarEntries ++ docJarEntries)
   }
 
   private def doPublish(
     builds: Seq[Build.Successful],
+    docBuilds: Seq[Build.Successful],
     workingDir: os.Path,
     logger: Logger
   ): Either[BuildException, Unit] = either {
 
+    assert(docBuilds.isEmpty || docBuilds.length == builds.length)
+
+    val it = builds.iterator.zip {
```

Comment:
what is `it` ? Maybe `buildsWithDocs`?

Reply:
`it` stands for a (short lived) iterator. `buildsWithDocsIt` could have worked too.